### PR TITLE
Restructure and rename admin controller

### PR DESCRIPTION
**Pull request by michaelKaefer**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Rename and restructure according to https://github.com/PrestaShop/prestafony-project/blob/master/docs/migration-guide.md under headline "Modern/Symfony Controllers"
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | -

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9159)

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
You're right... but as we have already published this page in 1.7.3 (I guess?) I don't know if we have the right to move the classes this way... ping @eternoendless.

Edit: to synchronize your repository with PrestaShop one, assuming the remote name of PrestaShop repository is "ps" and your own is "origin":

```
git checkout develop
git fetch ps
git rebase -i ps/develop
git push origin develop # now your repository is synced with our own
```

If you need the name of the remotes, try `git remote -v` :)

Edit2: don't be surprised I've just cleaned up your branch.

**Review by mickaelandrieu — DISMISSED** [review]

**Comment by michaelKaefer** [comment]
@mickaelandrieu thank you!

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
The new name is redundant, `Modules\ModuleCatalog`, can be a simple `Modules\Catalog`. It's like having `Grid\GridRow`, `Grid\GridColumn` which can be renamed in `Grid\Row` and `Grid\Column`.
But I approved the restructuring! :+1:

**Comment by mickaelandrieu** [comment]
@PierreRambaud make sense! One more update @michaelKaefer :) 2 things are really hard in programming: managing the cache, and naming things 🤣

**Comment by mickaelandrieu** [comment]
Edit: thinking about it, should we update [ThemeCatalogController](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php) to CatalogController into "Design" namespace @PierreRambaud?

**Comment by PierreRambaud** [comment]
Yep good idea :)

**Comment by mickaelandrieu** [comment]
One more "First contribution" issue then, don't do it!

**Comment by michaelKaefer** [comment]
@mickaelandrieu What is a "First contribution" issue?

**Comment by michaelKaefer** [comment]
@PierreRambaud @mickaelandrieu I removed the redundant name, thank you

File: src/PrestaShopBundle/Controller/Admin/Improve/Modules/ModuleCatalogController.php [hunk 1]
```diff
@@ -24,7 +24,7 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
-namespace PrestaShopBundle\Controller\Admin\Improve;
+namespace PrestaShopBundle\Controller\Admin\Improve\Modules;
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Improve/Modules/ModuleCatalogController.php]: You forgot to rename file name :) still ModuleCatalogController.php :)

**Comment by michaelKaefer** [comment]
@PierreRambaud Thank you! I fixed this stupid mistake of mine!

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Hum, look like there is something weird with commits list. Did you target the good branch?

**Comment by michaelKaefer** [comment]
@PierreRambaud That was me. Hope I can fix it next monday!

**Comment by PierreRambaud** [comment]
I'm closing this Pull request since the source is an unknow repository and this is in waiting for rebase since one year.

**Pull request closed without merging**